File: Leet_Code_Problems/1.two-sum.java
import java.util.HashMap;

/*
 * @lc app=leetcode id=1 lang=java
 *
 * [1] Two Sum
 */

// @lc code=start
class Solution {
    public int[] twoSum(int[] nums, int target) {

        int i, len = nums.length, diff = 0;
        HashMap<Integer, Integer> hashMap = new HashMap<>();
        int arr[] = new int[2];

        for (i = 0; i < len; i++) {
            diff = target - nums[i];
            if (hashMap.containsKey(diff)) {
                arr[1] = i;
                arr[0] = hashMap.get(diff);
                break;
            }
            hashMap.put(nums[i], i);
        }

        return arr;
    }
}
// @lc code=end

/* 
     KEY -->  Value of Array 
     VALUE -->  Index
*/

File: Leet_Code_Problems/1.two-sum.cpp
/*
 * @lc app=leetcode id=1 lang=cpp
 *
 * https://leetcode.com/problems/two-sum/
 *
 * [1] Two Sum
 */

// @lc code=start
class Solution
{
public:
    vector<int> twoSum(vector<int> &nums, int target)
    {

        unordered_map<int, int> hashMap;
        int i, len = nums.size(), rem = 0;

        for (i = 0; i < len; i++)
        {
            rem = target - nums[i];
            if (hashMap.find(rem) != hashMap.end()) //  found
                return {i, hashMap[rem]};

            else
                hashMap[nums[i]] = i;
        }

        return {};
    }
};
// @lc code=end
